### PR TITLE
HW-275: v2 optimisations

### PR DESCRIPTION
**Pull request by coldrunKacper**
V2 optimisations include:

- Excluding Activities with 'is_test' other than '0' and excluding 'RichTextEditor' custom fields.
- Removing debug messages.
- Moving Activity data loading code into separate function.
- Loading Activities by date when Activities number is greater than 5000.
- Implementing frontend date filter.
- Adding frontend CRM messages.
- Removing data caching on backend.

### CODE REVIEW

File: CRM/Activityreport/Data.php [hunk 1]
```diff
@@ -151,6 +151,17 @@ protected static function populateMultiValuesRow(array $row, array $fields, $off
     $found = true;
     $i = 0;
 
+    /*
+     * Check how many combinations of one $row needs to be created.
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: you can separate this into new private method : 

`
$info['multiValuesTotal'] = $this->getTotalCombinations($fields)
`

File: CRM/Activityreport/Data.php [hunk 2]
```diff
@@ -26,30 +26,24 @@ public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0) {
...
-        'return' => implode(',', array_keys(self::$fields)),
-        'options' => array(
-          'sort' => 'activity_date_time DESC',
-          'offset' => $offset,
-          'limit' => $limit,
-        ),
-      );
 
-      $activities = civicrm_api3('Activity', 'get', $params);
+    $params = array(
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: any reason for removing caching ?

Comment (coldrunKacper) [inline CRM/Activityreport/Data.php]: Yeah, cache was removed intentionally as it's going to be implemented in another phase (v3) after everything other is finished.

File: CRM/Activityreport/Data.php [hunk 3]
```diff
@@ -188,6 +177,27 @@ protected static function populateMultiValuesRow(array $row, array $fields, $off
...
   /**
+   * Get number of multivalues combinations for given Activity row.
+   *
+   * @param array $row
+   *   Activity row
+   * @param array $fields
+   *   Array containing all Activity fields
+   * @return int
+   */
+  protected static function getTotalCombinations(array $row, array $fields) {
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: I don't see a need for this method to be protected , private should be enough, same apply to the rest of the class properties and methods I don't see any need for them to be protected  for now and all should be private

**Pull request merged**